circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/Register.scala

```diff
@@ -8,6 +8,22 @@ import io.circe.Encoder
 object Register {
   type Registration[T] = (ru.TypeTag[T], (Encoder[T], Decoder[T]))
 
+  /**
+   * This method takes three implicit arguments: [[scala.reflect.api.TypeTags.TypeTag]], [[io.circe.Encoder]] and [[io.circe.Decoder]]. TypeTag is provided by the compiler,
+   * Encoder and Decoder are derived using Magnolia.
+   * There are several scenarios, in which derivation may fail, requiring defining custom codecs in a separate trait.
+   *
+   * Type class derivation will fail if the type or any of its fields don't have custom-defined Encoder/Decoder and at least one of the following statements about any of them is true:
+   *  - is a non-sealed trait
+   *  - is a sealed trait but two or more subtypes have the same name (in different packages)
+   *  - is a non-case class
+   *  - is a private case class
+   *  - has a private or with different signature apply method
```

Comment:
By `different` you mean `custom defined (as opposed to autogenerated)`?

Reply:
No

Reply:
If the custom defined on the surface looks like autogenerated, then it will work.

Reply:
Ok so maybe:
```suggestion
   *  - its `apply` method is private or has a different signature from what the autogenerated `apply` would have
```